pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -675,22 +640,15 @@ func (gce *GCECloud) CreateDisk(
 
 	mc := newDiskMetricContextZonal("create", gce.region, zone)
 
-	createOp, err := gce.manager.CreateDiskOnCloudProvider(
+	err = gce.manager.CreateDiskOnCloudProvider(
 		name, sizeGb, tagsStr, diskType, zone)
 
 	if isGCEError(err, "alreadyExists") {
 		glog.Warningf("GCE PD %q already exists, reusing", name)
 		return nil
-	} else if err != nil {
+	} else {
```

Comment:
Suggest not hiding the error occurrence from the metric. Just mc.observe(err) earlier.

Reply:
We don't consider this an actual error. Because we don't GET the disk before creating it, we rely on this error to tell us a disk exists from a previous use.

One concern @nicksardo brought up is that GCE errors could get spammy, but I think we are OK here. `CreateDisk` is only called as part of dynamic provisioning, after a PVC is created and no existing PVs match.  If a disk with the same name already exists, provisioning is considered successful, so GCE reports at most 1 error. In most cases there shouldn't be a disk collision because the provisioner generates a disk name that's unlikely to collide with disks used for other purposes, and the name is unique per PVC. It's unlikely that the bound PV disappears while the PVC and underlying disk still exist - the PV must've been manually deleted in that case.

Reply:
I think @nicksardo is saying to record the metric whether there is an error or not. At least that's what I understand from here https://github.com/kubernetes/kubernetes/pull/62297#discussion_r192485201

Reply:
Ah yeah thanks for the catch!